sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpan.java

```diff
@@ -418,22 +418,37 @@ private void addTimedEvent(EventData timedEvent) {
 
   @Override
   public ReadWriteSpan setStatus(StatusCode statusCode, @Nullable String description) {
-    if (statusCode == null) {
-      return this;
-    }
-    synchronized (lock) {
-      if (!isModifiableByCurrentThread()) {
-        logger.log(Level.FINE, "Calling setStatus() on an ended Span.");
-        return this;
-      } else if (this.status.getStatusCode() == StatusCode.OK) {
-        logger.log(Level.FINE, "Calling setStatus() on a Span that is already set to OK.");
-        return this;
+      if (statusCode == null) {
+          return this; // No action if statusCode is null
       }
-      this.status = StatusData.create(statusCode, description);
-    }
-    return this;
+      synchronized (lock) {
```

Comment:
We use [spotless](https://github.com/open-telemetry/opentelemetry-java/blob/main/CONTRIBUTING.md#style-guideline) to ensure consist code formatting. Please run `./gradlew spotlessApply`.